api-call.py

```diff
@@ -1,29 +1,226 @@
-import requests
+import datetime
 import json
 
+import pytz
+import requests
+import yaml
+
+
+def load_from_config():
+    """Loading important parameters from configuration file
+
+    :returns: The config object
+    """
+
+    with open('configuration.yaml', 'r') as file:
+        return yaml.safe_load(file)
+
+
+def log_error(error_message):
+    """Logging any error messages in the file and appending
+    with date time information in PST
+
+    :param error_message: The error message to log in file comes as a string
+    """
+
+    # Get the current datetime in PST
+    pst_timezone = pytz.timezone('America/Los_Angeles')
+    current_time = datetime.datetime.now(pst_timezone)
+    formatted_message = f"[{current_time}] {error_message}"
+
+    # Append the error message to the logfile
+    with open('logfile.txt', 'a') as file:
+        file.write(formatted_message + '\n')
```

Comment:
Most languages have corresponding logging packages and you don't have to reinvent the wheel: https://docs.python.org/3/library/logging.html
The current logging issue you might encounter is that the `logfile.txt` can become very big and you don't have a rotation policy implemented. It is good enough for a learning script though. Could you still look into these documents and compare them with your current logging approach? No need to rewrite your code this time, but feel free to do so if you want :)

Reply:
logging package used.